.buildkite/scripts/resolve_es_treeish.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
```

Comment:
I suggest we switch this to `set -euo pipefail`.

- `-u`: will fail fast is a var is undefined e.g. `ELASTIC_STACK_VERSION` (I was caught by this while running the script in isolated fashion)
- `-o pipefail`: since we are using pipes e.g. `| jq -r ...` this will catch errors with `jq` and fail fast.
-